### PR TITLE
fix(rust): improve error message listing a stopped node

### LINKED ISSUE #5146: Display a user-friendly error message when listing resources on a stopped node
### Observed behavior

The `list` commands should show a more descriptive error message when the node that is being queried is not running.

### Steps to reproduce

```sh
ockam node create n
ockam service list --at n # works as expected
ockam node stop n
ockam service list --at n # throws the error below

× connection peer was not found (origin: Transport, kind: Misuse, source location: /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/convert/mod.rs:727:9)
```

### Expected behavior

The error message should be user-friendly. Something like: `The node n is not running`.

### Ockam Version

0.89

### Proposed solution

Right after parsing the node name, check whether the node is running or not:

```rust
let node_name = parse_node_name(&node_name)?;
if !opts.state.nodes.get(&node_name)?.is_running() {
    return Err(miette!("The node {node_name} is not running"));
}
```

### Commands to fix

If you want to help with this issue, feel free to pick one of the commands below and apply the fix!

- [ ] `ockam node list`
- [ ] `ockam service list`
- [ ] `ockam tcp listener list`
- [ ] `ockam credential list`
- [ ] `ockam identity list`
- [ ] `ockam lease list`
- [ ] `ockam policy list`
- [ ] `ockam project list`
- [ ] `ockam project addon list`
- [ ] `ockam relay list`
- [ ] `ockam secure-channel list`
- [ ] `ockam space list`
- [ ] `ockam trust-context list`
- [ ] `ockam vault list`
- [ ] `ockam worker list`
- [ ] `ockam kafka-consumer list`
- [ ] `ockam kafka-producer list`

### PR DESCRIPTION
**Pull request by BradLewis**
## Current behavior

Currently when you try to list a stopped node, you get the obscure error
```shell
$ ockam service list --at my_stopped_node
OCK500 connection peer was not found (origin: Transport, kind: Misuse, source location: /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/convert/mod.rs:727:9)

       Please report this issue, with a copy of your logs, to https://github.com/build-trust/ockam/issues
       Version 0.89.0, hash: 11a2724331ecd51a1640a543930dd61a940a9a24
```

## Proposed changes

Will now check the error to see if it is due to the node being stopped. If it is, will instead return the message.

```shell
$ ockam service list --at my_stopped_node
  × The node 'my_stopped_node' is not running

         Version 0.89.0, hash: 1ba8e7b57506c9e2853c14187fd23e123e2e62a2

```

Issue is here https://github.com/build-trust/ockam/issues/5146

Note this will also slightly change the error for when a user requests a node that doesn't exist from 

```shell
$ ockam service list --at my_nonexistent_node
OCK500 not found

       Please report this issue, with a copy of your logs, to https://github.com/build-trust/ockam/issues
       Version 0.89.0, hash: 11a2724331ecd51a1640a543930dd61a940a9a24

``` 

to

```shell
$ ockam service list --at my_nonexistent_node
  × not found

         Version 0.89.0, hash: 4c5009ac04d11da27d9286e2930a91ce1f2f1fe7
``` 

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/service/list.rs [hunk 1]
```diff
@@ -67,21 +68,31 @@ async fn run_impl(
...
-    )?;
-    let json = serde_json::to_string_pretty(&services.list).into_diagnostic()?;
-    opts.terminal
-        .stdout()
-        .plain(plain)
-        .json(json)
-        .write_line()?;
-
-    Ok(())
+    let result = try_join!(send_req, progress_output);
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/service/list.rs]: We have a small utilty function to detect whether a node is running or not. Right after line 45 you can use the following snippet:

```rust
let node_name = parse_node_name(&node_name)?;
if !opts.state.nodes.get(&node_name)?.is_running() {
    return Err(miette!("The node {node_name} is not running"));
}
```

That way you don't have to check for specific errors/error messages.

**Review by adrianbenavides — COMMENTED** [review]
Hey @BradLewis, thanks for submitting this PR! I've left a comment about how to detect if a node is not running querying the node's state. Hope it helps, and let me know if you have any question!

File: implementations/rust/ockam/ockam_command/src/service/list.rs (hunk 1 above)
Comment (BradLewis) [reply implementations/rust/ockam/ockam_command/src/service/list.rs]: Ah that's much nicer! Thanks I'll make those changes.

**Comment by BradLewis** [comment]
I believe I've updated all of the `list` commands that I could find now. 
I also squashed all my commits into one this time given the changes were pretty small, but going forward, do you guys prefer  a single commit or just many smaller commits?

**Comment by BradLewis** [comment]
Hi Adrian, I have a couple of quick questions regarding the changes you made on the linked issue.
1. Is there anything I can do regarding the build issue? Seems to just be a library that needs updating.
2. I've updated all the commands that support the `--at` flag, however some of the commands listed, such as `ockam node list` don't support it, so I'm not sure there's anything needed to be done there.
3. Currently I don't parse the node name through the `parse_node_name` function due to it being fallible, and I didn't want to change existing functionality where possible, and some functions use the non-parsed node name. Would it be better to run the name through `parse_node_name` before checking if a node is running?

Thanks!

**Comment by adrianbenavides** [comment]
> Hi Adrian, I have a couple of quick questions regarding the changes you made on the linked issue.
> 
> 1. Is there anything I can do regarding the build issue? Seems to just be a library that needs updating.

Nope, don't worry about that, we are looking into that already and it's external to this PR.

> 2. I've updated all the commands that support the `--at` flag, however some of the commands listed, such as `ockam node list` don't support it, so I'm not sure there's anything needed to be done there.

You're correct, I added all the list commands, but only those with the `--at` flag are affected by the bug described at the issue.

> 3. Currently I don't parse the node name through the `parse_node_name` function due to it being fallible, and I didn't want to change existing functionality where possible, and some functions use the non-parsed node name. Would it be better to run the name through `parse_node_name` before checking if a node is running?

Ideally, we'd like to use the `parse_node_name` in all cases so that the user can write `/node/n` or `n` as the argument. So, if you have spotted any command where we are not doing it already, it would awesome if you apply that fix as well.

**Comment by BradLewis** [comment]
> Ideally, we'd like to use the `parse_node_name` in all cases so that the user can write `/node/n` or `n` as the argument. So, if you have spotted any command where we are not doing it already, it would awesome if you apply that fix as well.

Perfect, thanks for the clarifications. I'll make those changes after work this evening.

**Comment by adrianbenavides** [comment]
> > Ideally, we'd like to use the `parse_node_name` in all cases so that the user can write `/node/n` or `n` as the argument. So, if you have spotted any command where we are not doing it already, it would awesome if you apply that fix as well.
> 
> Perfect, thanks for the clarifications. I'll make those changes after work this evening.

Whenever you can, thanks! 🙏

**Comment by BradLewis** [comment]
I added `parse_node_name` for the `tcp-listener list --at` command as it was missing, however for the rest of them that aren't using `parse_node_name`, they're using the function `extract_address_value` which as far as I can tell does basically the same thing but supports more than just nodes. Not sure if we want to update them to just use `parse_node_name` instead since `--at` only supports nodes, but if not then all should be good here.

**Comment by adrianbenavides** [comment]
> they're using the function `extract_address_value` which as far as I can tell does basically the same thing but supports more than just nodes.

That's exactly right, but don't worry about that. I think that's a little out of the scope for this PR and we can tackle that anytime.

**Comment by adrianbenavides** [comment]
One last thing before we can merge your PR, we require all commits to be signed as described here:

> All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.

Once you have your keys setup, you will have to (probably squash and) force-push to sign your commits. If you need any help let me know!

**Comment by BradLewis** [comment]
> One last thing before we can merge your PR, we require all commits to be signed as described here:

Thanks, I'm not sure why these previous commits weren't signed but that's fixed now. I've also squashed the 2 commits into 1.

Edit: Ah actually it might be due to me rebasing the branch onto develop through the PR itself rather than the CLI. Either way it's fixed now.

**Review by adrianbenavides — APPROVED** [review]
Ready to merge! Thanks again @BradLewis for helping out with this and congrats on your first contribution to Ockam!

**Pull request merged**